main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -460,10 +460,20 @@ object Weeder2 {
         traverse(fields)(visitStructField)
       ) {
         (doc, ann, mods, ident, tparams, fields) =>
-        // TODO: Validation, e.g., never duplicate names
-        Validation.success(Declaration.Struct(
-          doc, ann, mods, ident, tparams, fields.sortBy(_.loc), tree.loc
+        // Ensure that each name is unique
+        val errors = getDuplicates(fields, (f: StructField) => f.ident.name)
+        // For each field, only keep the first occurrence of the name
+        val groupedByName = fields.groupBy(_.ident.name)
+        val filteredFields = groupedByName.values.map(_.head).toList
+        val result = Validation.success(Declaration.Struct(
```

Comment:
You can inline this local result variable. That way there is no confusion.

Reply:
In fact, its probably more clean todo:

if (errors.isEmpty) 
 ...
else 
 ...

so you handle the positive case first.